maa-cli/src/dirs.rs

```diff
@@ -734,4 +880,54 @@ mod tests {
         #[cfg(windows)]
         ensure_name("foo\\bar");
     }
+
+    #[test]
+    fn test_cleanup() {
+        let _version = match run::core_version() {
+            Ok(v) => v,
+            Err(_) => return, // uninitialized
+        };
+
+        fn test(paths: &Vec<&Path>, targets: &Vec<CleanupTarget>) {
+            let files: Vec<PathBuf> = paths
+                .iter()
+                .map(|path| {
+                    let file = if path.is_dir() {
+                        path.join("file.txt")
+                    } else {
+                        path.to_path_buf()
+                    };
+                    let _ = std::fs::File::create(&file);
+                    file
+                })
+                .collect();
+            let _result = cleanup(targets);
+
+            for file in files {
+                assert!(!file.exists());
+                let _ = std::fs::remove_file(file);
+            }
+        }
+
+        test(&vec![cache()], &vec![CleanupTarget::CliCache]);
+        test(
+            &vec![&log().join("asst.log"), &log().join("asst.bak.log")],
+            &vec![CleanupTarget::Log],
+        );
+        test(
+            &vec![&state().join("cache").join("avatars")],
+            &vec![CleanupTarget::Avatars],
+        );
+
+        let debug = log();
+        test(
+            &vec![
+                &debug.join("drops"),
+                &debug.join("map"),
+                &debug.join("other"),
+                &debug.join("Roguelike"),
+            ],
+            &vec![CleanupTarget::Map],
+        );
```

Comment:
BTW, this test should be ignored, due to write to user space.

Reply:
I will try to use a trait so that it's path can be overwritten in test cases.
`trait PathProvider {
    fn get_path(&self) -> Vec<PathBuf>;
}`

Reply:
I'm not sure if you misunderstand, the `ignored` is an attribute of rust, which mark tests as ignored and will not be executed by default unless pass `--include-ignored` in `cargo test`. Tests writing to user space is Ok, but we need to avoid to run it in some case.